game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/retreat/DefensiveSubsRetreatTest.java

```diff
@@ -118,41 +129,200 @@ void hasNoNamesWhenCanNotSubmergeAndNoRetreatTerritories() {
     assertThat(defensiveSubsRetreat.getNames(), is(empty()));
   }
 
-  @Test
-  void retreatHappensWhenNotSubmersibleButHasRetreatTerritories() {
-    final Territory retreatTerritory = mock(Territory.class);
-    when(retreatTerritory.isWater()).thenReturn(true);
-    when(retreatTerritory.getUnitCollection()).thenReturn(mock(UnitCollection.class));
-
-    final BattleState battleState =
-        givenBattleStateBuilder()
-            .defendingUnits(List.of(givenUnitCanEvade()))
-            .battleSite(battleSite)
-            .gameData(
-                givenGameData()
-                    .withTerritoryHasNeighbors(battleSite, Set.of(retreatTerritory))
-                    .build())
-            .build();
-    final DefensiveSubsRetreat defensiveSubsRetreat =
-        new DefensiveSubsRetreat(battleState, battleActions);
-
-    assertThat(defensiveSubsRetreat.getNames(), hasSize(1));
+  @Nested
+  class SubmergeHappens {
+
+    @Mock IDisplay display;
+    @Mock ISound sound;
+    @Mock IDelegateHistoryWriter historyWriter;
+
+    @BeforeEach
+    public void setupMocks() {
+      when(delegateBridge.getDisplayChannelBroadcaster()).thenReturn(display);
+      when(delegateBridge.getSoundChannelBroadcaster()).thenReturn(sound);
+      when(delegateBridge.getHistoryWriter()).thenReturn(historyWriter);
+      when(defender.getName()).thenReturn("defender");
+    }
+
+    @Test
+    void retreatHappensWhenHasNoRetreatTerritoriesButIsSubmersible() {
+      final GameData gameData = MockGameData.givenGameData().withSubmersibleSubs(true).build();
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'MockGameData.givenGameData' due to existing static import 'games.strategy.triplea.delegate.battle.steps.MockGameData.givenGameData'](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6208307)